packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -677,52 +735,52 @@ export class IntervalCollectionView<TInterval extends ISerializableInterval> ext
         intervalType: MergeTree.IntervalType,
         props?: MergeTree.PropertySet,
     ) {
-        let seq = 0;
-        if (this.client) {
-            seq = this.client.getCurrentSeq();
+        const interval: TInterval = this.localCollection.addInterval(start, end, intervalType, props, true);
+
+        if (interval) {
+            const serializedInterval = {
+                end,
+                intervalType,
+                properties: interval.properties,
+                sequenceNumber: this.client?.getCurrentSeq() ?? 0,
+                start,
+            };
+            // Local ops get submitted to the server. Remote ops have the deserializer run.
+            this.emitter.emit("add", undefined, serializedInterval);
         }
 
-        const serializedInterval: ISerializedInterval = {
-            end,
-            intervalType,
-            properties: props,
-            sequenceNumber: seq,
-            start,
-        };
+        this.emit("addInterval", interval, true, undefined);
 
-        return this.addInternal(serializedInterval, true, undefined);
+        return interval;
     }
 
     public delete(
         start: number,
-        end: number) {
-        let sequenceNumber = 0;
-        if (this.client) {
-            sequenceNumber = this.client.getCurrentSeq();
+        end: number,
+        local: boolean = true,
+        op?: ISequencedDocumentMessage) {
```

Comment:
this doesn't look right. we shouldn't be exposing op or local on public methods

Reply:
this goes for a number of methods. the public api surface should be abstracted from the internals of dealing with remote ops.

Reply:
I see. I'll refactor the delete path so this doesn't happen. You're not talking about, for instance, 


<!--StartFragment-->public addInternal(        serializedInterval: ISerializedInterval, local: boolean, op: ISequencedDocumentMessage)<!--EndFragment-->

, right? For LocalCollection.addInterval, I passed in "local" to control whether we create a new ID. If we got an interval over the wire without an ID, I thought it would be best to leave it in that state, rather than have each client create its own ID. But maybe that's not a case we need to worry about, so maybe LocalCollection.addInterval can just create an ID if it doesn't see one in the properties.